lib/oli_web/live/products/details/image_upload.ex

```diff
@@ -0,0 +1,128 @@
+                </div>
+              </div>
+            </div>
+          </section>
+
+            <section class="row mb-2" id="img-preview">
+            {#if @uploads.cover_image.entries != []}
+              {#for entry <- @uploads.cover_image.entries}
+                <article class="upload-entry col-12">
+                  {#if !entry_has_errors?(@uploads.cover_image, entry)}
+                  <figure>
+                    { live_img_preview entry, class: "img-fluid w-75"}
+                    <figcaption class="text-muted">{ entry.client_name }</figcaption>
+                  </figure>
+
+                  <div class="row d-flex">
+                    <div class="col-8 align-self-center h-100">
+                      <div class="progress">
+                        <div role="progressbar" class="progress-bar w-100" style={"width:#{entry.progress} !important"} value={entry.progress} max="100" aria-valuemin="0" aria-valuemax="100"> { entry.progress }% </div>
+                      </div>
+                    </div>
+                    <div class="col-4 align-self-center h-100">
+                      <button class="btn btn-secondary btn-sm" phx-click="cancel_upload" phx-value-ref={entry.ref} aria-label="cancel">&times;</button>
+                    </div>
+                  </div>
+
+                  {/if}
+
+                  {#for err <- upload_errors(@uploads.cover_image, entry)}
+                    <div class="alert danger">{ upload_error(err) }</div>
+                  {/for}
+
+                </article>
+              {/for}
+            {#else}
+                <article class="col-12">
+                    <img id="current-product-img" src={@product.cover_image} class="img-fluid w-75" />
+                </article>
+            {/if}
+            </section>
+
+            <Submit class="btn btn-primary mt-3" label="Save" opts={ [disabled: !upload_has_entries?(@uploads.cover_image) or upload_has_errors?(@uploads.cover_image)] }/>
+          </Form>
+        </div>
+      </div>
+
+    </div>
+
+    <script type="text/javascript">
+      $(document).ready(function(){
+        $('#drag-and-drop-zone').bind('dragover', function(){
+          $(this).addClass('on-drag');
+        });
+        $('#drag-and-drop-zone').bind('dragleave', function(){
+          $(this).removeClass('on-drag');
+        });
+      });
+    </script>
+    """
+  end
+
+  defp upload_has_entries?(upload) do
+    upload.entries != []
+  end
+
+  defp upload_has_errors?(upload) do
+    Enum.any?(upload.entries, &entry_has_errors?(upload, &1))
+  end
+
+  defp entry_has_errors?(upload, entry) do
+    upload_errors(upload, entry) != []
+  end
+
+  defp upload_error(:too_large), do: "Image too large, try again with a image lower than 5 MB."
+  defp upload_error(:too_many_files), do: "Too many files, try again with a single file"
+  defp upload_error(:not_accepted), do: "Unacceptable file type, try again with a .jpg .jpeg or .png file"
```

Comment:
Wdyt about adding one more to match everything in case the error is not in this list?